rspace/src/main/scala/coop/rchain/rspace/history/HistoryInstances.scala

```diff
@@ -341,27 +341,142 @@ object HistoryInstances {
           elements.lastOption.map(t => (remainingPath, t))
         )
 
+      def processSubtree(
+          start: (Blake2b256Hash, Option[LastModification])
+      )(elem: (Byte, List[HistoryAction])): F[(Byte, (Blake2b256Hash, Option[(KeyPath, Trie)]))] =
+        elem match {
+          case (b, actions) =>
+            actions
+              .foldLeftM(start) {
+                case (
+                    (currentRoot, previousModificationOpt),
+                    InsertAction(remainingPath, value)
+                    ) =>
+                  insert(currentRoot, previousModificationOpt, remainingPath, value)
+
+                case (
+                    (currentRoot, previousModificationOpt),
+                    DeleteAction(remainingPath)
+                    ) =>
+                  delete(currentRoot, previousModificationOpt, remainingPath)
+              }
+              .map((b, _))
+        }
+
+      def transformSubtreeResults(elem: (Byte, Trie)): Seq[(Byte, Seq[Byte], NonEmptyTriePointer)] =
+        elem match {
+          case (_, EmptyTrie)                               => Nil
+          case (b, Skip(prefix, ptr)) if (b == prefix.head) => (b, prefix.tail.toSeq, ptr) :: Nil
+          case (_, _: Skip)                                 => Nil //this feels like an error
```

Comment:
then raise it! Let's risk blowing up for a valid input once or twice instead of building on invalid/dubious state indefinitely :)

Reply:
good catch!

Reply:
the comment was outdated, added proper ones.

Reply:
> the comment was outdated
duh... aren't they always?